slice-mapper/src/main/java/com/cognifide/slice/mapper/impl/processor/ChildrenFieldProcessor.java

```diff
@@ -95,4 +111,53 @@ private Object getArrayFromList(Class<?> componentType, List<?> children) {
 		}
 		return array;
 	}
+
+	private List<?> getRemoteChildrenList(Resource resource, ValueMap valueMap, Field field,
+			String propertyName) {
+
+		final Class<?> fieldType = field.getType();
+		Object value = valueMap.get(propertyName);
+
+		if (value == null) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug(
+						"the property [{}/{}] is undefined, assigning null value for [{}#{}]",
+						new Object[] { resource.getPath(), propertyName,
+								fieldType.getCanonicalName(), field.getName() });
+			}
+			return Collections.EMPTY_LIST;
+		}
+
+		if (!(value instanceof String)) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug(
+						"the property [{}/{}] annotated is not of String type as required by " +
+								"@JcrProperty(..., follow = true) in the model, assigning null " +
+								"value for [{}#{}]",
+						new Object[] { resource.getPath(), propertyName, fieldType.getCanonicalName(),
+								field.getName() });
+			}
+			return Collections.EMPTY_LIST;
+		}
+
+		String nestedResourcePath = (String) value;
+		Resource followUpResource = resource.getResourceResolver().getResource(nestedResourcePath);
+
+		if (followUpResource == null) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug(
+						"the nested resource [{}/{}] expected under path [{}] doesn't exist, " +
+								"assigning null value for [{}#{}]",
+						new Object[] { resource.getPath(), propertyName, nestedResourcePath,
+								fieldType.getCanonicalName(), field.getName() });
+			}
+			return Collections.EMPTY_LIST;
+		}
```

Comment:
The logic of getting followUpResource is the same for ChildrenFieldProcessor and SliceResourceFieldProcessor. Please extract it to some class and reference here and in SRFP.